packages/babel-plugin-transform-block-scoping/src/index.ts

```diff
@@ -207,43 +222,62 @@ const letReferenceFunctionVisitor = traverse.visitors.merge([
     },
   },
   tdzVisitor,
-]);
+] as Visitor<LetReferenceVisitorState>[]);
 
-const hoistVarDeclarationsVisitor = {
+const hoistVarDeclarationsVisitor: Visitor<BlockScoping> = {
   enter(path, self) {
-    const { node, parent } = path;
-
     if (path.isForStatement()) {
-      if (isVar(node.init, node)) {
-        const nodes = self.pushDeclar(node.init);
+      const { node } = path;
+      if (isVar(node.init)) {
+        const nodes = self.pushDeclar(node.init as t.VariableDeclaration);
         if (nodes.length === 1) {
           node.init = nodes[0];
         } else {
           node.init = t.sequenceExpression(nodes);
         }
       }
-    } else if (path.isFor()) {
-      if (isVar(node.left, node)) {
-        self.pushDeclar(node.left);
-        node.left = node.left.declarations[0].id;
+    } else if (path.isForInStatement() || path.isForOfStatement()) {
+      const { node } = path;
+      if (isVar(node.left)) {
+        self.pushDeclar(node.left as t.VariableDeclaration);
```

Comment:
If we change the return type of `isVar` to `node is t.VariableDeclaration`, can we avoid these type casts?

Reply:
indeed, that looks to be a better option